server/routerlicious/packages/lambdas/src/scribe/summaryWriter.ts

```diff
@@ -591,19 +607,24 @@ export class SummaryWriter implements ISummaryWriter {
                     $lt: lt,
                 },
             };
-            const logTail = await this.opStorage.find(query, { "operation.sequenceNumber": 1 });
-
-            // If the db is not updated with all logs yet, get them from checkpoint messages.
-            if (logTail.length !== (lt - gt - 1)) {
-                const nextSeq = logTail.length === 0 ? gt : logTail[logTail.length - 1].operation.sequenceNumber + 1;
-                for (const message of pending) {
-                    if (message.operation.sequenceNumber >= nextSeq && message.operation.sequenceNumber < lt) {
-                        logTail.push(message);
-                    }
+
+            const logTailOpMessage = await this.opStorage.find(query, { "operation.sequenceNumber": 1 });
+            logTail = logTailOpMessage.map((log) => log.operation);
+        }
+        
+        Lumberjack.info(`LogTail of length ${logTail.length} fetched from seq no ${gt} to ${lt}`, this.lumberProperties);
+
+        // If the db is not updated with all logs yet, get them from checkpoint messages.
+        if (logTail.length !== (lt - gt - 1)) {
+            Lumberjack.info(`Populating logtail gaps`, this.lumberProperties);
```

Comment:
(Unrelated to this PR) A more reliable implementation seems to be
1. union logTail and pending
2. dedupe and order the union'd set
3. filter the set based on gt and lt

Reply:
Correct..this implementation seems quite messy. It is not even clear to me if the fallback to this code is happening often..I added a log line to track how often this code path is getting exercised..will modify/remove it in the next PR as appropriate